pre_commit_hooks/ros_include_guard.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2022 Tier IV, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import pathlib
+
+
+class DirectiveBase:
+
+  def __init__(self, directive):
+    self.directive = directive
+    self.line = None
+    self.text = None
+    self.expected = None
+    self.tokens = None
+
+  def update(self, line, text):
+    self.line = line
+    self.text = text
+
+  def is_none(self):
+    return self.line is None
+
+  def mismatch(self):
+    return self.text != self.expected
+
+  def overwrite(self, lines):
+    lines[self.line] = self.expected
+
+
+class OpeningDirective(DirectiveBase):
+
+  def prepare(self, macro_name):
+    self.tokens = split_space_boundary(self.text)
+    if 3 <= len(self.tokens):
+      self.tokens[2] = macro_name
+      self.expected = ''.join(self.tokens)
+    else:
+      self.expected = F'{self.directive} {macro_name}'
+
+
+class ClosingDirective(DirectiveBase):
+
+  def prepare(self, macro_name):
+    self.tokens = split_space_boundary(self.text)
+    if 5 <= len(self.tokens):
+      self.tokens[4] = macro_name
+      self.expected = ''.join(self.tokens)
+    else:
+      self.expected = F'{self.directive}  // {macro_name}'
+
+
+class IncludeGuard:
+
+  def __init__(self):
+    self.ifndef = OpeningDirective('#ifndef')
+    self.define = OpeningDirective('#define')
+    self.endif  = ClosingDirective('#endif')
+    self.pragma = False
+
+  def items(self):
+    yield self.ifndef
+    yield self.define
+    yield self.endif
+
+  def is_none(self):
+    return any(item.is_none() for item in self.items())
+
+  def mismatch(self):
+    return any(item.mismatch() for item in self.items())
+
+  def overwrite(self, lines):
+    return any(item.overwrite(lines) for item in self.items())
+
+  def prepare(self, macro_name):
+    for item in self.items(): item.prepare(macro_name)
+
+
+def split_space_boundary(text, delimiters={' '}):
+
+  result = []
+  prev = None
+  for char in text:
+    curr = char in delimiters
```

Comment:
Caught by the spell-check.
```suggestion
    current = char in delimiters
```

Reply:
Fixed. https://github.com/tier4/pre-commit-hooks-ros/pull/32/commits/a65eb9b838df4e7132f7a967cee0a25f87ca46a2